plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2733,7 +2733,7 @@ public String stopVM(final Connect conn, final String vmName, final boolean forc
         if (forceStop) {
             return stopVMInternal(conn, vmName, true);
         }
-        String ret = stopVM(conn, vmName, false);
+        String ret = stopVMInternal(conn, vmName, false);
         if (ret == Script.ERR_TIMEOUT) {
             ret = stopVMInternal(conn, vmName, true);
         } else if (ret != null) {
```

Comment:
if it is not force stop then any timeouts or resource exception should be sent to the user. I think that is the valid legacy behaviour. Now if there is a timeout instead of throwing exception the action will automatically get converted to force shutdown @rhtyd @borisstoyanov